File: connectors/connector-github/src/test/java/com/taskadapter/connector/github/GithubToGTaskTest.java
package com.taskadapter.connector.github;

import com.taskadapter.connector.testlib.TestDataLoader;
import com.taskadapter.model.AllFields;
import org.eclipse.egit.github.core.Issue;
import org.junit.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class GithubToGTaskTest {
    @Test
    public void convertsBasicTask() {
        var issue = (Issue) TestDataLoader.load("issue.json", Issue.class);
        var task = GithubToGTask.toGtask(issue);
        assertThat(task.getValue(AllFields.summary)).isEqualTo("task 1");
    }
}

File: connectors/connector-github/src/test/scala/com/taskadapter/connector/github/GithubToGTaskTest.scala
package com.taskadapter.connector.github

import com.taskadapter.connector.testlib.TestDataLoader
import com.taskadapter.model.Summary
import org.eclipse.egit.github.core.Issue
import org.junit.Assert.assertEquals
import org.junit.runner.RunWith
import org.scalatest.junit.JUnitRunner
import org.scalatest.{FunSpec, Matchers}

@RunWith(classOf[JUnitRunner])
class GithubToGTaskTest extends FunSpec with Matchers {
  it("converts a basic task") {
    val issue = TestDataLoader.load("issue.json", classOf[Issue]).asInstanceOf[Issue]
    val task = GithubToGTask.toGtask(issue)
    assertEquals("task 1", task.getValue(Summary))
  }
}